js/main.js

```diff
@@ -12,16 +12,101 @@ $photoUrl.addEventListener('input', updateSRC);
+  dataEntry.entryId = data.nextEntryId;
   data.entries.unshift(dataEntry);
   data.nextEntryId++;
   $form.reset();
   $previewPhoto.src = 'images/placeholder-image-square.jpg';
+  $newEntryForm.setAttribute('class', 'hidden');
+  $entriesList.setAttribute('class', 'entries-list');
+  data.view = 'entries';
 }
-$form.addEventListener('submit', retrieveData);
+$form.addEventListener('submit', storeData);
+
+function prependDOM(event) {
+  $entriesUnorderedList.prepend(createEntry(data.entries[0]));
+}
+$form.addEventListener('submit', prependDOM);
+
+// entries list generation code
+var $entriesList = document.querySelector('.entries-list');
+var $entriesUnorderedList = document.querySelector('ul');
+var $emptyList = document.querySelector('.empty-entry-list');
+
+function createEntry(entry) {
+  $emptyList.setAttribute('class', 'hidden');
+  var $newEntry = document.createElement('li');
+  $newEntry.setAttribute('class', 'entry-container full-flex');
+
+  var $newEntryImageContainer = document.createElement('div');
+  $newEntryImageContainer.setAttribute('class', 'column-half image-container');
+  var $newEntryImage = document.createElement('img');
+  $newEntryImage.setAttribute('src', entry.pictureLink);
+  $newEntryImageContainer.appendChild($newEntryImage);
+  $newEntry.appendChild($newEntryImageContainer);
+
+  var $newEntryTextDiv = document.createElement('div');
+  $newEntryTextDiv.setAttribute('class', 'column-half entry-text');
+
+  var $newEntryTitle = document.createElement('h2');
+  var $newEntryTitleText = document.createTextNode(entry.title);
+  $newEntryTitle.setAttribute('class', 'entry-title-text');
+  $newEntryTitle.appendChild($newEntryTitleText);
+  $newEntryTextDiv.appendChild($newEntryTitle);
+
+  var $newEntryNotes = document.createElement('p');
+  var $newEntryNotesText = document.createTextNode(entry.notes);
+  $newEntryNotes.setAttribute('class', 'entry-notes-text');
+  $newEntryNotes.appendChild($newEntryNotesText);
+  $newEntryTextDiv.appendChild($newEntryNotes);
+
+  $newEntry.appendChild($newEntryTextDiv);
+  return $newEntry;
+}
+
+function generateDOM(event) {
+  for (var dataLength = 0; dataLength < data.entries.length; dataLength++) {
+    $entriesUnorderedList.appendChild(createEntry(data.entries[dataLength]));
+  }
+}
+document.addEventListener('DOMContentLoaded', generateDOM);
+
+// button to new entry form
+function goToNewForm(event) {
+  $newEntryForm.setAttribute('class', 'new-entry-form');
+  $entriesList.setAttribute('class', 'hidden');
+  data.view = 'entry-form';
+}
+var $newEntryButton = document.querySelector('.new-entry-button');
+$newEntryButton.addEventListener('click', goToNewForm);
+
+// nav links
+var $entriesLink = document.querySelector('.header-entries-link');
+function goToEntriesViaLink(event) {
+  $newEntryForm.setAttribute('class', 'hidden');
+  $entriesList.setAttribute('class', 'entries-list');
+  data.view = 'entries';
+}
+$entriesLink.addEventListener('click', goToEntriesViaLink);
```

Comment:
You would benefit from making a single function named something like `switchView` that receives an argument of the name of the view to switch to, and the function should show that view and hide all other views. That way, you won't need to make a new function for each new view created.